internal/helpers/secrets_test.go

```diff
@@ -393,6 +400,37 @@ func TestSyncSecret(t *testing.T) {
 					"not the owner of the destination Secret foo/baz")
 			},
 		},
+		{
+			name:               "invalid-dest-exists-no-create-overwrite",
+			client:             clientBuilder.Build(),
+			obj:                ownerWithDestNoCreateClobber,
+			createDest:         false,
+			expectSecretsCount: 0,
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				return assert.EqualError(t, err,
+					"destination secret foo/baz does not exist, and create=false")
+			},
+		},
+		{
+			name:               "dest-exists-not-owned-overwrite-true",
+			client:             clientBuilder.Build(),
+			obj:                ownerWithDestClobber,
+			createDest:         true,
+			expectSecretsCount: 1,
+			wantErr:            assert.NoError,
+		},
+		{
+			name:               "dest-exists-owned-overwrite-true",
+			client:             clientBuilder.Build(),
+			obj:                ownerWithDestClobber,
+			createDest:         true,
+			destLabels:         OwnerLabels,
+			expectSecretsCount: 1,
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				return assert.ErrorContains(t, err,
+					"not the owner of the destination Secret foo/baz")
+			},
```

Comment:
Does the operator check the type of object that owns a Secret? Just thinking it might be good to allow overwriting ownership if the owner is a non-VSO object, to help when migrating to VSO. Or I suppose it could allow multiple owner references, as long as only one is the VSO object being reconciled.

Reply:
I believe with this change VSO will only take ownership of the destination secret if it does not have the generic VSO owner labels.

Allowing other Kinds to own the secret is an interesting idea, but I am not sure how that would work in the case where the "other" Kind is deleted - VSO would probably resolve that on the next reconciliation.